src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,346 @@
+                            tempList.add(
+                                    number.replaceAll(REGROUPED_PHONE_NUMBER_PATTERN,
+                                            "$1$2$3$4$5$6".concat("\\".concat(twoDigits.replaceAll("^\\d", specialChar))).concat("$8$9")));
+                        }));
+
+        wrongPhoneNumbers.clear();
+        wrongPhoneNumbers.addAll(tempList);
+
+        return wrongPhoneNumbers;
+    }
+
+    private static List<String> createCorrectUserNames() {
+        List<String> correctNames = new ArrayList<>(List.of(
+                "pushkin",
+                "ThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100SymbolsThisNameHasLengthMoreThan100Symbols",
+                "1"));
+        List<String> specialChars = getSpecialCharacters("");
+        specialChars.addAll(getRussianAndUkrLetters());
+
+        correctNames.addAll(specialChars);
+
+        return correctNames;
+    }
+
+    private static List<String> createWrongUserNames() {
+
+        return List.of(
+                "",
+                " ",
+                "aaa bbb",
+                " aaa",
+                "aaa ");
+    }
+
+    private static List<String> createCorrectWordsEmptyOrMin2Max50() {
+        List<String> correctWords = new ArrayList<>(List.of(
+                "aZ",
+                "99",
+                "Za0",
+                "wW1-",
+                "wW2-w",
+                "wW3'",
+                "wW4'w",
+                "wW5`",
+                "wW6`w",
+                "wW7_",
+                "wW8_w"
+        ));
+
+        List<String> tempList = new ArrayList<>();
+        correctWords.forEach(word -> getRussianAndUkrLetters().forEach(
+                letter -> tempList.add(word.concat(letter))
+        ));
+
+        correctWords.addAll(tempList);
+        correctWords.addAll(List.of(
+                "",
+                "LengthIs50LengthIs50LengthIs50LengthIs50LengthIs50"));
+
+        return correctWords;
+    }
+
+    private static List<String> createWrongWordsEmptyOrMin2Max50() {
+        String correctWord = "a1A";
+
+        List<String> wrongWords = new ArrayList<>(List.of(
+                "w",
+                "W",
+                "1",
+                " ",
+                " 1",
+                " W",
+                " w",
+                "1 ",
+                "1 ",
+                "W ",
+                "w ",
+                "ThisWordHasLengthMoreThan50CharactersItsLengthIs51S"));
+
+        List<String> specialChars = getSpecialCharacters("");
+        specialChars.removeAll(List.of("_", "-", "`", "'"));
+
+        specialChars.forEach(specialChar -> wrongWords.add(correctWord.concat(specialChar)));
+
+        return wrongWords;
+    }
+
+    private static List<String> getSpecialCharacters(String escapedSymbol) {
+        List<String> specialCharacters = new ArrayList<>();
+
+        IntStream.range(33, 48).forEach(i -> specialCharacters.add(escapedSymbol + ((char) i)));
```

Comment:
Nice job! But it could be improved as well. Create a new method that will be getting int beginning, int end and add characters to passed list.